packages/message-relayer/src/exec/withdraw.ts

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env ts-node
+
+/**
+ * Utility that will relay all L2 => L1 messages created within a given L2 transaction.
+ */
+
+/* Imports: External */
+import { ethers } from 'ethers'
+import { predeploys, getContractInterface } from '@eth-optimism/contracts'
+import { sleep } from '@eth-optimism/core-utils'
+import dotenv from 'dotenv'
+
+/* Imports: Internal */
+import { getMessagesAndProofsForL2Transaction } from '../relay-tx'
+
+dotenv.config()
+const l1RpcProviderUrl = process.env.WITHDRAW__L1_RPC_URL
+const l2RpcProviderUrl = process.env.WITHDRAW__L2_RPC_URL
+const l1PrivateKey = process.env.WITHDRAW__L1_PRIVATE_KEY
+const l1StateCommitmentChainAddress =
+  process.env.WITHDRAW__STATE_COMMITMENT_CHAIN_ADDRESS
+const l1CrossDomainMessengerAddress =
+  process.env.WITHDRAW__L1_CROSS_DOMAIN_MESSENGER_ADDRESS
+
+const main = async () => {
+  const l2TransactionHash = process.argv[2]
+  if (l2TransactionHash === undefined) {
+    throw new Error(`must provide l2 transaction hash`)
+  }
+
+  const l1RpcProvider = new ethers.providers.JsonRpcProvider(l1RpcProviderUrl)
+  const l1Wallet = new ethers.Wallet(l1PrivateKey, l1RpcProvider)
+  const l1WalletBalance = await l1Wallet.getBalance()
+  console.log(`relayer address: ${l1Wallet.address}`)
+  console.log(`relayer balance: ${ethers.utils.formatEther(l1WalletBalance)}`)
+
+  const l1CrossDomainMessenger = new ethers.Contract(
+    l1CrossDomainMessengerAddress,
+    getContractInterface('OVM_L1CrossDomainMessenger'),
+    l1Wallet
+  )
+
+  console.log(`searching for messages in transaction: ${l2TransactionHash}`)
+  let messagePairs = []
+  while (true) {
+    try {
+      messagePairs = await getMessagesAndProofsForL2Transaction(
+        l1RpcProviderUrl,
+        l2RpcProviderUrl,
+        l1StateCommitmentChainAddress,
+        predeploys.OVM_L2CrossDomainMessenger,
+        l2TransactionHash
+      )
+      break
+    } catch (err) {
+      if (err.message.includes('unable to find state root batch for tx')) {
```

Comment:
This isn't covered with tests so a change in the error message could break this downstream. Probably better to do something like this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#handling_a_specific_error

But not the biggest deal